spyder/plugins/application/widgets/install.py

```diff
@@ -0,0 +1,274 @@
+        reply.setStandardButtons(QMessageBox.Yes | QMessageBox.No)
+        reply.exec_()
+        if reply.result() == QMessageBox.Yes:
+            self.start_installation_update()
+        else:
+            self._change_update_installation_status(status=PENDING)
+
+    def finished_installation(self, status):
+        """Handle finished installation."""
+        if status == FINISHED or status == PENDING:
+            self.setup()
+            self.accept()
+
+    def close_installer(self):
+        """Close the installation dialog."""
+        if (self.status == FINISHED
+                or self.status == CANCELLED):
+            self.setup()
+            self.accept()
+        else:
+            self.hide()
+
+    def reject(self):
+        """Reimplemented Qt method."""
+        on_installation_widget = self._installation_widget.isVisible()
+        if on_installation_widget:
+            self.close_installer()
+        else:
+            super(UpdateInstallerDialog, self).reject()
+
+    def _change_update_installation_status(self, status=NO_STATUS):
+        """Set the installation status."""
+        logger.debug(f"Installation status: {status}")
+        self.status = status
+        self.sig_installation_status.emit(self.status)
+
+    def _progress_reporter(self, block_number, read_size, total_size):
+        progress = 0
+        if total_size > 0:
+            progress = block_number * read_size
+        if self.cancelled:
+            raise UpdateInstallationCancelledException()
+        else:
+            self.sig_download_progress.emit(progress, total_size)
+
+    def cancel_thread_install_update(self):
+        self._change_update_installation_status(status=CANCELLED)
+        self.thread_install_update.join()
+
+    def _download_install(self):
+        try:
+            logger.debug("Downloading installer executable")
+            tmpdir = tempfile.mkdtemp(prefix="Spyder-")
+            is_full_installer = (is_module_installed('numpy') or
+                                 is_module_installed('pandas'))
+            if os.name == 'nt':
+                name = 'Spyder_64bit_{}.exe'.format('full' if is_full_installer
+                                                    else 'lite')
+            else:
+                name = 'Spyder{}.dmg'.format('' if is_full_installer
+                                             else '-Lite')
+
+            url = ('https://github.com/spyder-ide/spyder/releases/latest/'
+                   f'download/{name}')
+            destination = os.path.join(tmpdir, name)
+            logger.debug(f"Downloading installer from: {url}")
+            download = urlretrieve(url,
+                                   destination,
+                                   reporthook=self._progress_reporter)
+            self._change_update_installation_status(status=INSTALLING)
+            cmd = ('start' if os.name == 'nt' else 'open')
+            subprocess.run([cmd, destination], shell=True)
```

Comment:
Seems that the command syntax was incorrect. `shell=True` requires a single string command; `shell=False` requires a list of command elements.
We can either use
```suggestion
            subprocess.run(' '.join([cmd, destination]), shell=True)
```
Or
```suggestion
            subprocess.run([cmd, destination])
```
Can you confirm that these also work for Windows?

Reply:
Checked on Windows and I think we will need to go with the option that still uses `shell=True` (since on Windows an error - [WinError 2] - raises without it). Could you apply the suggestion that adds the `' '.join` call @jsbautista ? 

Also, thanks for checking @mrclary !

Reply:
As I can confirm in my testing, `start` is a shell-level command, so we do need the shell to execute it. However, we should really try to avoid `shell=True` if at all possible, as it is less performant, less reliable and there are [security issues](https://docs.python.org/3.12/library/subprocess.html#security-considerations) with (though this case is probably safe—though I can't figure out where `destination` comes from here. Is there any reason we need to use `start` from the shell here instead of just executing what we need directly with a fully qualified path?

Reply:
I think that was the way @jsbautista found to be able to launch the installer and keep it running even when it forces Spyder to exit (on the Windows case where the NSIS installer validates if Spyder is running and closes it if the user agrees)